elyra/pipeline/processor_airflow.py

```diff
@@ -236,29 +235,27 @@ def _cc_pipeline(self, pipeline, pipeline_name):
                 # Retrieve component from cache
                 component = self._component_registry.get_component(operation.classifier)
 
-                # Change value of variables according to their type. String variables must include
-                # quotation marks in order to render properly in the jinja template and dictionary
-                # values must be converted from strings.
+                # Convert the user-entered value of certain properties according to their type
                 for component_property in component.properties:
                     # Skip properties for which no value was given
                     if component_property.ref not in operation.component_params.keys():
                         continue
+
+                    # Get corresponding property's value from parsed pipeline
+                    property_value = operation.component_params.get(component_property.ref)
+
+                    self.log.debug(f"Processing component parameter '{component_property.name}' "
+                                   f"of type '{component_property.type}'")
                     if component_property.type == "string":
-                        # Get corresponding component_property value from parsed pipeline and convert
-                        op_property = operation.component_params.get(component_property.ref)
-                        operation.component_params[component_property.ref] = json.dumps(op_property)
-                    elif component_property.type in ['dict', 'dictionary']:
-                        # Get corresponding component_property value from parsed pipeline and convert
-                        op_property = operation.component_params.get(component_property.ref)
-                        if not op_property:
-                            op_property = "{}"
-                        operation.component_params[component_property.ref] = ast.literal_eval(op_property)
-                    elif component_property.type in ['list', 'set', 'array', 'arr']:
-                        op_property = operation.component_params.get(component_property.ref)
-                        # Get corresponding component_property value from parsed pipeline and convert
-                        if not op_property:
-                            op_property = "[]"
-                        operation.component_params[component_property.ref] = ast.literal_eval(op_property)
+                        # Add surrounding quotation marks to string value for correct rendering
+                        # in jinja DAG template
+                        operation.component_params[component_property.ref] = json.dumps(property_value)
+                    elif component_property.type == 'dictionary':
+                        processed_value = operation.process_dictionary_value(property_value, logger=self.log)
+                        operation.component_params[component_property.ref] = processed_value
```

Comment:
Now that we may end up returning a string value, I'll have to add some additional logic here to put surrounding quotation marks around strings such that they are rendered properly in the Airflow DAG

Reply:
Interesting.  If that's the case (i.e., the need for runtime-specific differences), then we should probably move these kinds of methods from `Operation` to `RuntimePipelineProcessor` and let the airflow processor first call the superclass, then apply any additional quotations.  (Assuming I'm understanding the context)

Reply:
Ah that is a good point! Let me just double check that there isn't anything I'm overlooking, and if not, I'll make that change